build-support/common.sh

```diff
@@ -48,31 +48,17 @@ function git_merge_base() {
 }
 
 function determine_python() {
-  if [[ -n "${PY:-}" ]]; then
-    which "${PY}" && return 0
+  version = '3.9'
+  interpreter_path="$(command -v "python${version}")"
```

Comment:
It would be good to still use `PY` if it is set... it seems fine not to validate it if someone has asked for a particular interpreter (i.e., an early return is ok).

Reply:
I agree with Stu because it can be useful to force Pants to use a particular version of Python 3.9

Reply:
For example: Josh and I were experimenting with Pypy and nogil Pythons recently.